pandas/io/formats/latex.py

```diff
@@ -10,56 +11,91 @@
+class RowStringConverter:
+    r"""Converter for dataframe rows into LaTeX strings.
 
     Parameters
     ----------
     formatter : `DataFrameFormatter`
-    column_format : str, default None
-        The columns format as specified in `LaTeX table format
-        <https://en.wikibooks.org/wiki/LaTeX/Tables>`__ e.g 'rcl' for 3 columns
-    longtable : boolean, default False
-        Use a longtable environment instead of tabular.
+        Instance of `DataFrameFormatter`.
+    multicolumn: bool, optional
+        Whether to use \multicolumn macro.
+    multicolumn_format: str, optional
+        Multicolumn format.
+    multirow: bool, optional
+        Whether to use \multirow macro.
 
-    See Also
-    --------
-    HTMLFormatter
     """
 
     def __init__(
         self,
         formatter: DataFrameFormatter,
-        column_format: Optional[str] = None,
-        longtable: bool = False,
         multicolumn: bool = False,
         multicolumn_format: Optional[str] = None,
         multirow: bool = False,
-        caption: Optional[str] = None,
-        label: Optional[str] = None,
-        position: Optional[str] = None,
     ):
         self.fmt = formatter
         self.frame = self.fmt.frame
-        self.bold_rows = self.fmt.bold_rows
-        self.column_format = column_format
-        self.longtable = longtable
         self.multicolumn = multicolumn
         self.multicolumn_format = multicolumn_format
         self.multirow = multirow
-        self.caption = caption
-        self.label = label
-        self.escape = self.fmt.escape
-        self.position = position
-        self._table_float = any(p is not None for p in (caption, label, position))
+        self.clinebuf: List[List[int]] = []
+        self.strcols = self._get_strcols()
+        self.strrows = list(zip(*self.strcols))
+
+    def get_strrow(self, row_num: int) -> str:
+        """Get string representation of the row."""
+        row = self.strrows[row_num]
+
+        is_multicol = row_num < self._clevels and self.fmt.header and self.multicolumn
+
+        is_multirow = (
+            row_num >= self._nlevels
+            and self.fmt.index
+            and self.multirow
+            and self._ilevels > 1
+        )
+
+        is_cline_maybe_required = is_multirow and row_num < len(self.strrows) - 1
+
+        crow = self._preprocess_row(row)
+
+        if is_multicol:
+            crow = self._format_multicolumn(crow)
+        if is_multirow:
+            crow = self._format_multirow(crow, row_num)
+
+        lst = []
+        lst.append(" & ".join(crow))
+        lst.append(" \\\\")
+        if is_cline_maybe_required:
+            cline = self._compose_cline(row_num, len(self.strcols))
+            lst.append(cline)
+        return "".join(lst)
+
+    @property
+    def _header_row_num(self):
+        """Number of rows in header."""
+        return self._nlevels if self.fmt.header else 0
+
+    @property
+    def _ilevels(self):
```

Comment:
i would make more descriptive properties here; they don't really need to be private, but ok

e.g.

index_nlevels is more descriptive.

Reply:
Actually, it was not clear for me what these ``_ilevels``-like values were in the original code. So, I decided not to change the names. As you suggested, I renamed them.